csrc/multidevice/pipeline_ir.h

```diff
@@ -0,0 +1,128 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+#include <disjoint_set.h>
+#include <fusion.h>
+#include <ir/base_nodes.h>
+#include <multidevice/device_mesh.h>
+#include <multidevice/multidevice.h>
```

Comment:
Could  you pull execution information out of pipeline definition, and remove these includes?

Reply:
Done, thanks! I also remove the inclusion of `fusion.h`